flutter/lib/ui/geometry.dart

```diff
@@ -680,36 +680,188 @@ class Rect {
   String toString() => "Rect.fromLTRB(${left.toStringAsFixed(1)}, ${top.toStringAsFixed(1)}, ${right.toStringAsFixed(1)}, ${bottom.toStringAsFixed(1)})";
 }
 
-// A rounded rectangle with the same radii for all four corners.
+/// A radius for either circular or oval shapes.
+class Radius {
+  /// Constructs a circular radius. [x] and [y] will have the same radius value.
+  const Radius.circular(double radius) : this.oval(radius, radius);
+
+  /// Constructs an oval radius with the given radii.
+  const Radius.oval(this.x, this.y);
+
+  /// A radius with the value zero.
+  ///
+  /// You can use [Radius.zero] with [RRect] to have right-angle corners.
+  static const Radius zero = const Radius.circular(0.0);
+
+  /// The radius value on the horizontal axis.
+  final double x;
+
+  /// The radius value on the vertical axis.
+  final double y;
+
+  /// Linearly interpolate between two radii.
+  ///
+  /// If either is null, this function substitutes [Radius.zero] instead.
+  static Radius lerp(Radius a, Radius b, double t) {
+    if (a == null && b == null)
+      return null;
+    if (a == null)
+      return new Radius.oval(b.x * t, b.y * t);
+    if (b == null) {
+      double k = 1.0 - t;
+      return new Radius.oval(a.x * k, a.y * k);
+    }
+    return new Radius.oval(
+      lerpDouble(a.x, b.x, t),
+      lerpDouble(a.y, b.y, t)
+    );
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other is! Radius)
+      return false;
+    final Radius typedOther = other;
+    return typedOther.x == x && typedOther.y == y;
+  }
+
+  @override
+  int get hashCode => hashValues(x, y);
+
+  @override
+  String toString() {
+    return x == y ? 'Radius.circular(${x.toStringAsFixed(1)})' :
+        'Radius.oval(${x.toStringAsFixed(1)}, ${y.toStringAsFixed(1)})';
+  }
+}
+
+/// A rounded rectangle with the custom radii for all four corners.
 class RRect {
   RRect._();
 
   /// Construct a rounded rectangle from its left, top, right, and bottom edges,
-  /// and the radii along its horizontal axis and its vertical axis.
-  RRect.fromLTRBXY(double left, double top, double right, double bottom, double radiusX, double radiusY) {
+  /// and the same radii along its horizontal axis and its vertical axis.
+  RRect.fromLTRBXY(double left, double top, double right, double bottom,
+                   double radiusX, double radiusY) {
     _value
       ..[0] = left
       ..[1] = top
       ..[2] = right
       ..[3] = bottom
       ..[4] = radiusX
-      ..[5] = radiusY;
+      ..[5] = radiusY
+      ..[6] = radiusX
+      ..[7] = radiusY
+      ..[8] = radiusX
+      ..[9] = radiusY
+      ..[10] = radiusX
+      ..[11] = radiusY;
   }
 
-  /// Construct a rounded rectangle from its bounding box and the radii along
-  /// its horizontal axis and its vertical axis.
+  /// Construct a rounded rectangle from its left, top, right, and bottom edges,
+  /// and a radius that is the same in each corner.
```

Comment:
and a radius that is the same in each corner => and the same radius in each corner